prometheus/promhttp/instrument_server.go

```diff
@@ -438,30 +435,30 @@ func (r *responseWriterDelegator) Write(b []byte) (int, error) {
 	return n, err
 }
 
-type fancyResponseWriterDelegator struct {
+type classicFancyResponseWriterDelegator struct {
 	*responseWriterDelegator
 }
 
-func (r *fancyResponseWriterDelegator) Status() int {
+func (r *classicFancyResponseWriterDelegator) Status() int {
```

Comment:
Thinking about it, this should be forwarded automatically from the embedded responseWriterDelegator, i.e. you should not need to implement this method.